src/runner/browser-job.js

```diff
@@ -27,13 +25,16 @@ export default class BrowserJob extends EventEmitter {
 
         this.completionQueue = [];
 
-        this.connectionErrorListener = error => this._setResult(RESULT.errored, error);
-
+        this.connectionErrorListener = error => {
```

Comment:
Is it necessary?

Reply:
There is no difference with the previous code.

Reply:
To properly await `_setResult` in a case of error, you'll need to store a promise of `_setResult` execution when it's invoked first time, and return that Promise on subsequent `_setResult` calls. In this case `this._setResult(RESULT.errored, ...)` call will be awaited in the `BrowserJob#abort` method when trying to call `this._setResult(RESULT.aborted)`.